neuro_san/internals/run_context/interfaces/agent_network_inspector.py

```diff
@@ -0,0 +1,49 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+from typing import Any
+from typing import Dict
+
+
+class AgentNetworkInspector:
+    """
+    Interface describing an entity that can examine the configuration of the
+    agent network spec.
+    Having this interface breaks some circular dependencies.
+    """
```

Comment:
This is the interface that RunContexts can use to ask questions about the agent network spec.